app/controllers/api/json/imports_controller.rb

```diff
@@ -83,6 +83,42 @@ def create
     end
   end
 
+  # -------- Cancel a running import (using update PUT request) -------
+
+  def update
+    if !params[:item_queue_id].present?
+      render_jsonp({ errors: { imports: 'Missing queue ID in import cancellation request'} }, 400) and return
+    end
+
+    queue_id = params[:item_queue_id]
+
+    begin
+      # first verify that this job exists and user owns this job
+      import = DataImport.where(id: queue_id).first()
+
+      if import.nil? or import.user_id != current_user.id
+        render_jsonp({ errors: { imports: "Failed to cancel import. Job with ID #{queue_id} not found."} }, 400) and return
+      end
+
+      if import.state == Carto::DataImport::STATE_COMPLETE or import.state == Carto::DataImport::STATE_FAILURE
+        render_jsonp({ errors: { imports: "Failed to cancel import. Job with ID #{queue_id} not currently running."} }, 400) and return
+      end
+
+      # found a running job for the user with the given ID; stop it
+      Resque::Job.destroy(Resque::ImporterJobs, job_id: queue_id)
```

Comment:
I'm not sure if this is actually an issue, but since the import job processes continue to run even after cancellation, I'm wondering if there could be a situation where the job process creates new database objects after it has already been cancelled. Those objects would hang around since the cancellation function would have already been run by the time the job finishes, so I was thinking it might be good for the `runner.rb` to double check that its job id still exists, and if it doesn't, undo anything that it managed to accomplish after the cancellation.